Tools/cases_generator/generate_cases.py

```diff
@@ -70,25 +72,37 @@ def write(
         # Write cache effect variable declarations
         cache_offset = 0
         for ceffect in self.cache_effects:
-            if ceffect.name != "unused":
-                # TODO: if name is 'descr' use PyObject *descr = read_obj(...)
-                bits = ceffect.size * 16
-                f.write(f"{indent}    uint{bits}_t {ceffect.name} = ")
-                if ceffect.size == 1:
-                    f.write(f"*(next_instr + {cache_offset});\n")
+            if ceffect.name != UNUSED:
+                bits = ceffect.size * BITS_PER_CODE_UNIT
+                if bits == 64:
+                    # NOTE: We assume that 64-bit data in the cache
+                    # is always an object pointer.
+                    # If this becomes false, we need a way to specify
+                    # syntactically what type the cache data is.
+                    f.write(
+                        f"{indent}    PyObject *{ceffect.name} = "
+                        f"read_obj(next_instr + {cache_offset});\n"
+                    )
                 else:
-                    f.write(f"read_u{bits}(next_instr + {cache_offset});\n")
+                    f.write(f"{indent}    uint{bits}_t {ceffect.name} = ")
+                    if ceffect.size == 1:
+                        # There is no read_u16() helper function.
+                        f.write(f"*(next_instr + {cache_offset});\n")
+                    else:
```

Comment:
Eh, should we just add a `read_u16` function to keep things simple? They're not really special anymore now that we've ditched the structs, and we've shown that it would compile the same.

Reply:
Oh, good idea. Will do.